doc/advanced-testing-mini-course/src/Lec04FaultInjection.lhs

```diff
@@ -182,50 +187,104 @@ Model
 >     ( m { mModel = Vector.snoc (mModel m) bs }
 >     , Just (WriteResp (Index (Vector.length (mModel m))))
 >     )
->   ClientRequest (ReadReq (Index ix)) -> (m, Just (ReadResp (mModel m Vector.!? ix)))
+>   ClientRequest (ReadReq (Index ix)) -> (m, Just (ReadResp (mModel m Vector.! ix)))
 >   InjectFault fault -> (m { mMaybeFault = Just fault}, Nothing)
 >   RemoveFault       -> (m { mMaybeFault = Nothing}, Nothing)
 >   Reset             -> (m, Nothing)
 
-> exec :: Command -> IORef (Maybe Fault) -> Manager -> IO (Maybe ClientResponse)
+> data Result a = Ok a | Fail | Info | Nemesis
+>   deriving Show
+
+> exec :: Command -> IORef (Maybe Fault) -> Manager -> IO (Result ClientResponse)
 > exec (ClientRequest req) _ref mgr =
 >   case req of
->     WriteReq bs -> Just . WriteResp <$> httpWrite mgr bs
->     ReadReq ix  -> Just . ReadResp  <$> httpRead mgr ix
+>     WriteReq bs -> do
+>       res <- try (httpWrite mgr bs)
+>       case res of
+>         Left (err :: HttpException) | isResponseTimeout err -> return Info
+>                                     | otherwise             -> return Fail
```

Comment:
Maybe not needed here, but if it gives a 500 in status code it should also be `Info` . But we might not need that in this case.

Reply:
Fixed, thanks!